File: Chap.91.알고리즘수업/01.피보나치수/24416.py
def fib1(n):
    global cnt1
    if n == 1 or n == 2:
        cnt1 += 1
        return 1
    else:
        return fib1(n-1) + fib1(n-2)
    
def fib2(n):
    global cnt2
    f = [0, 1, 1] + [0] * (n - 2)
    for i in range(3, n + 1):
        cnt2 += 1
        f[i] = f[i-1] + f[i-2]
    return f[n]

N = int(input())    
cnt1, cnt2 = 0, 0
fib1(N)
fib2(N)
print(cnt1, cnt2)
    

File: Chap.56.피보나치/24416.피보나치수/24416.cpp
#include <bits/stdc++.h>
using namespace std;

int cnt1, cnt2;

int fib1(int n) {
    if (n == 1 || n == 2) {
        cnt1++; return 1;
    } else {
        return fib1(n-1) + fib1(n-2);
    }
}

int fib2(int n) {
    int a = 1, b = 1;
    for (int i = 3; i <= n; i++) {
        int t = a + b;
        a = b; b = t; cnt2++;
    }
    return b;
}

int main() {
    int n; cin >> n;
    fib1(n); fib2(n);
    cout << cnt1 << " " << cnt2 << endl;
}